### PR TITLE
[CB-53] Allocated Objects Strategy

### PR DESCRIPTION
**Pull request by EPecherkin**
[CB-53](https://youtrack.toptal.net/youtrack/issue/CB-53)

### CODE REVIEW

File: features/change_files_spec.rb [hunk 1]
```diff
@@ -16,29 +16,48 @@ class Class1
...
+    )
+  end
+
+  it 'generates map if Class1 reopen is changed' do
+    class1_reopen_path.open('w') { |f| f.write <<~RUBY }
+      class Class1
+      end
+    RUBY
+
+    is_expected.to include(
```
Comment (jaimerson) [inline features/change_files_spec.rb:36]: Why is it `include` now instead of `match_array`? Shouldn't it remain strict?

**Review by jaimerson — APPROVED** [review]
lgtm

File: features/change_files_spec.rb (hunk 1 above)
Comment (EPecherkin) [reply features/change_files_spec.rb:36]: I don't think so. It is not that bad if we predict a bit more. So, I think it is better to specify examples which should contain than to specify all output of predictor

Comment (jaimerson) [reply features/change_files_spec.rb:36]: Maybe, but these specs allow it to predict a lot more. Is the full output inconveniently longer?

**Review by pluff — APPROVED** [review]

**Comment by pluff** [comment]
Please squash everything into meaningful commits

**Pull request merged**